site/js/components/ee.js

```diff
@@ -0,0 +1,227 @@
+    localStorage.setItem('chromeos-m100-current-prompt', value);
+    this.curPrompt = value;
+  }
+
+  /**
+   * Clear the current prompt value.
+   * @return {void}
+   */
+  _clearCurrentPrompt() {
+    localStorage.removeItem('chromeos-m100-current-prompt');
+    this.curPrompt = 0;
+  }
+
+  /**
+   * Gets the current quiz start value from local storage.
+   * @return {int} - current quiz start value
+   */
+  _getQuizStart() {
+    const quizStart = !!localStorage.getItem('chromeos-m100-quiz-start');
+    return quizStart;
+  }
+
+  /**
+   * Sets the current quiz start value.
+   * @param {int} value - the current quiz start integer value
+   * @return {void}
+   */
+  _setQuizStart(value) {
+    localStorage.setItem('chromeos-m100-quiz-start', value);
+    this.quizStart = value;
+  }
+
+  /**
+   * Clear the current quiz start value.
+   * @return {void}
+   */
+  _clearQuizStart() {
+    localStorage.removeItem('chromeos-m100-quiz-start');
+    this.quizStart = false;
+  }
+
+  /**
+   * Prompt the question to the user
+   * @param {int} prompt
+   */
+  _promptQuestion(prompt = 0, ret = false) {
+    if (prompt < this.text.prompts.length && this.text.prompts[prompt].question.length > 0) {
+      if (ret) {
+        return this.text.prompts[prompt].question;
+      }
+      console.info('Question #%s: %c%s', prompt + 1, 'font-weight: bold;', this.text.prompts[prompt].question);
+    } else if ((prompt = this.text.prompts.length)) {
+      this._promptReward();
+    }
+  }
+
+  /**
+   * Prompt the hint to the user
+   * @param {int} prompt
+   */
+  _promptHint(prompt) {
+    if (this.text.prompts[prompt].hint.length > 0) {
+      console.info('Hint: %c%s', 'font-style: italic;', this.text.prompts[prompt].hint);
+    }
+  }
+
+  /**
+   * Prompt reward
+   * @param {int} prompt
+   */
+  _promptReward() {
+    console.info('%c%s', 'font-weight: bold; font-size: 1.25em;', this.text.reward.headline);
+    for (const iterator of this.text.reward.body) {
+      console.info(iterator);
+    }
+  }
+
+  /**
+   * Check the answer
+   * @param {int} prompt
+   * @param {string} value
+   * @return {void}
+   */
+  _checkAnswer(prompt, value) {
+    // console.info('Question #%s: %c%s', prompt + 1, 'font-weight: bold; font-style: italic;', this._promptQuestion(prompt, true));
+    if (value.toString().toLowerCase() == this.text.answers[prompt].toString().toLowerCase()) {
+      // Correct
+      console.info('You answered: %c%s', 'font-weight: bold; font-style: italic; background: #0d652d; color: white; padding: .25em;', value.toString());
+      console.info(this.text.prompts[prompt].responseCorrect);
+      this._setCurrentPrompt(prompt + 1);
+      if (this.curPrompt < this.text.prompts.length) {
+        console.info('Question #%s: %c%s', this.curPrompt + 1, 'font-weight: bold;', this._promptQuestion(this.curPrompt, true));
+      } else if ((this.curPrompt = this.text.prompts.length)) {
+        this._promptReward();
+      }
```

Comment:
This looks like it's duplicated from `this._promptQuestion`. Can this all just be replaced with a call to that instead?

Reply:
good catch. I had different text strings before, but normalized them.